src/main/java/assets/game.js

```diff
@@ -363,8 +361,35 @@ function place(size) {
     }
 }
 
+// More customizable highlighting. Takes in template position 0-indexed
+// length of highlight for vertical rotation.
+function place2D(pos, len) {
+    return function () {
+        let row = this.parentNode.rowIndex;
+        let col = this.cellIndex;
+        vertical = (document.getElementById("is_vertical").dataset.toggled) == "true";
+        let table = document.getElementById("player");
+
+        pos.forEach(function(cord) {
+            let row_off = cord[0];
+            let col_off = cord[1];
+            let cell;
+            if (vertical) {
+                let temp = row_off;
+                row_off = col_off;
+                col_off = len - temp - 3;
+            }
+
+            cell = table.rows[row+row_off].cells[col+col_off];
```

Comment:
During submarine placement, undefined errors are thrown since table.rows[row+row_off] is sometimes undefined. This causes the program to access the 'cells' variable which doesn't exist.

Reply:
Fix looks good.